test/lib/wovnrb_test.rb

```diff
@@ -108,8 +108,112 @@ def test_switch_lang_ignores_amp_defined_with_symbol_attribute
     assert_switch_lang('en', 'ja', body, body, false)
   end
 
+  def test_call_changes_environment_for_next_stack_call_without_path_ignored
+    settings = {
+      'project_token' => '123456',
+      'url_pattern' => 'path',
+      'default_lang' => 'ja',
+      'supported_langs' => ['ja', 'en'],
+      'ignore_paths' => ['/en/ignored']
+    }
+    env = {
+      'rack.input' => '',
+      'HTTP_HOST' => 'test.com',
+      'REQUEST_URI' => '/en/not_ignored',
+      'PATH_INFO' => '/en/not_ignored'
+    }
+
+    assert_call_affects_env(settings, env, true, true)
+  end
+
+  def test_call_changes_environment_for_next_stack_call_with_path_ignored_with_language_code
+    settings = {
+      'project_token' => '123456',
+      'url_pattern' => 'path',
+      'default_lang' => 'ja',
+      'supported_langs' => ['ja', 'en'],
+      'ignore_paths' => ['/en/ignored']
+    }
+    env = {
+      'rack.input' => '',
+      'HTTP_HOST' => 'test.com',
+      'REQUEST_URI' => '/ignored',
+      'PATH_INFO' => '/ignored'
+    }
+
+    assert_call_affects_env(settings, env, false, true)
+  end
+
+  def test_call_changes_environment_for_next_stack_call_with_path_ignored_without_language_code
+    settings = {
+      'project_token' => '123456',
+      'url_pattern' => 'path',
+      'default_lang' => 'ja',
+      'supported_langs' => ['ja', 'en'],
+      'ignore_paths' => ['/ignored']
+    }
+    env = {
+      'rack.input' => '',
+      'HTTP_HOST' => 'test.com',
+      'REQUEST_URI' => '/en/ignored',
+      'PATH_INFO' => '/en/ignored'
+    }
+
+    assert_call_affects_env(settings, env, false, true)
+  end
+
+  def test_call_changes_environment_for_next_stack_call_with_path_ignored_without_language_code_in_original_language
+    settings = {
+      'project_token' => '123456',
+      'url_pattern' => 'path',
+      'default_lang' => 'ja',
+      'supported_langs' => ['ja', 'en'],
+      'ignore_paths' => ['/ignored']
+    }
+    env = {
+      'rack.input' => '',
+      'HTTP_HOST' => 'test.com',
+      'REQUEST_URI' => '/ignored',
+      'PATH_INFO' => '/ignored'
+    }
+
+    assert_call_affects_env(settings, env, false, true)
+  end
+
+  def test_call_does_not_change_environment_for_next_stack_call_with_path_ignored
+    settings = {
+      'project_token' => '123456',
+      'url_pattern' => 'path',
+      'default_lang' => 'ja',
+      'supported_langs' => ['ja', 'en'],
+      'ignore_paths' => ['/en/ignored']
+    }
+    env = {
+      'rack.input' => '',
+      'HTTP_HOST' => 'test.com',
+      'REQUEST_URI' => '/en/ignored',
+      'PATH_INFO' => '/en/ignored'
+    }
+
+    assert_call_affects_env(settings, env, false, false)
+  end
+
   private
 
+  def assert_call_affects_env(settings, env, mock_api, affects)
```

Comment:
confused about the `affects` parameter. Isn't the 3rd parameter of assert_equal just a message?

Reply:
ah, it's actually the second parameter. I would maybe make these keyword parameters since seeing `false, false` in the test is hard to read what is being asserted

Reply:
> I would maybe make these keyword parameters since seeing false, false in the test is hard to read what is being asserted

agree, will do, thanks

Reply:
done, thanks